File: extract.py
import sys

with open('lib/features/swipe/ui/swipe_screen.dart', 'r') as f:
    lines = f.readlines()

card_start = -1
card_end = -1
button_start = -1
button_end = -1

for i, line in enumerate(lines):
    if 'class TinderSwipeCardStack' in line:
        card_start = i
    if '// ACTION BUTTON WIDGET' in line:
        card_end = i - 1
        button_start = i + 2
    if 'class _ActionButton ' in line:
        button_start = i

button_end = len(lines)

with open('lib/features/swipe/widgets/swipe_card.dart', 'w') as f:
    f.write("import 'package:flutter/material.dart';\n")
    f.write("import 'dart:math' as math;\n")
    f.write("import '../../../common/models/place_model.dart';\n")
    f.write("import 'swipe_buttons.dart';\n\n")
    for i in range(card_start, card_end):
        f.write(lines[i].replace('_ActionButton', 'ActionButton'))

with open('lib/features/swipe/widgets/swipe_buttons.dart', 'w') as f:
    f.write("import 'package:flutter/material.dart';\n\n")
    for i in range(button_start, button_end):
        f.write(lines[i].replace('_ActionButton', 'ActionButton'))

with open('lib/features/swipe/ui/swipe_screen.dart', 'w') as f:
    for i in range(0, card_start):
        f.write(lines[i])

print("Extraction successful")

File: extract.dart
import 'dart:developer' as developer;
import 'dart:io';

void main() {
  final file = File('lib/features/swipe/ui/swipe_screen.dart');
  final lines = file.readAsLinesSync();

  int cardStart = -1;
  int cardEnd = -1;
  int buttonStart = -1;

  for (int i = 0; i < lines.length; i++) {
    if (lines[i].contains('class TinderSwipeCardStack')) {
      cardStart = i;
    }
    if (lines[i].contains('// ACTION BUTTON WIDGET')) {
      cardEnd = i - 1;
      buttonStart = i + 2;
    }
    if (lines[i].contains('class _ActionButton ')) {
      buttonStart = i;
    }
  }

  int buttonEnd = lines.length;

  final cardFile = File('lib/features/swipe/widgets/swipe_card.dart');
  final cardContent = StringBuffer();
  cardContent.writeln("import 'package:flutter/material.dart';");
  cardContent.writeln("import 'dart:math' as math;");
  cardContent.writeln("import '../../../common/models/place_model.dart';");
  cardContent.writeln("import 'swipe_buttons.dart';\n");
  for (int i = cardStart; i < cardEnd; i++) {
    cardContent.writeln(lines[i].replaceAll('_ActionButton', 'ActionButton'));
  }
  cardFile.writeAsStringSync(cardContent.toString());

  final buttonFile = File('lib/features/swipe/widgets/swipe_buttons.dart');
  final buttonContent = StringBuffer();
  buttonContent.writeln("import 'package:flutter/material.dart';\n");
  for (int i = buttonStart; i < buttonEnd; i++) {
    buttonContent.writeln(lines[i].replaceAll('_ActionButton', 'ActionButton'));
  }
  buttonFile.writeAsStringSync(buttonContent.toString());

  final newScreenContent = StringBuffer();
  for (int i = 0; i < cardStart; i++) {
    newScreenContent.writeln(lines[i]);
  }
  file.writeAsStringSync(newScreenContent.toString());

  developer.log('Extraction successful');
}
